### DISCUSSION TITLE
Invisible U+200B characters in code copied to clipboard

### DISCUSSION THREAD

**Discussion by frak0d** [community/community#14860] [Mobile]
When i copy any part of code from GitHub app, there are some invisible U+200B characters in it, which are not present in source file

**Comment by stevepopovich** [comment]
Hi there! Thanks for the feedback. Do you know if this is happening on iOS or Android? Also what are the exact steps for to reproduce this bug?

**Reply by frak0d** [reply]
Don't know about iOS, but its definately happening On Android.

Steps:-
1. Open app.
2. Browse any repo.
3. View any file in repo.
4. Long press a line to select and then copy using copy button (in top right).

**Reply by stevepopovich** [reply]
Fair enough, I see the characters. Just so I understand fully, what does having these charcters in the copied code prevent you from doing?

**Reply by frak0d** [reply]
> what does having these charcters in the copied code prevent you from doing?

They cause compilation to fail. Both clang and gcc treat them as normal characters and not as whitespace. So having an invisible character adjacent to an identifier changes its meaning.


For example here is a invisible character before `using` :-
```
error: '\U0000200busing' was not declared in this scope
    6 |  ​using zh::literals;
      |  ^~~~~
```

**Reply by stevepopovich** [reply]
Oh okay. Trying to understand how you use this. You are pulling code out of repos and running them in code somewhere else then? and that breaks compilation?

**Reply by frak0d** [reply]
> Oh okay. Trying to understand how you use this.

I have one big private repository where i store all my reusable code snippets. I prefer not to clone the whole thing and only get the snippets that i currently need.

> You are pulling code out of repos and running them in code somewhere else then? and that breaks compilation?

Yes but The Reason For failed compilation is presence of those invisible Characters.

If i copy using GitHub website(in raw mode), then it works fine.

**Reply by stevepopovich** [reply]
Great. Interesting use case! This is definitely a bug and I will pass on to the team to get prioritized and fixed. Thanks for the detailed feedback!
